api-gateway/src/main/java/org/swasth/apigateway/filters/HCXValidationFilter.java

```diff
@@ -79,6 +79,9 @@ public GatewayFilter apply(Config config) {
                     jweRequest.validate(getMandatoryHeaders(), getSubject(exchange), timestampRange, senderDetails, recipientDetails);
                     jweRequest.validateUsingAuditData(allowedEntitiesForForward, allowedRolesForForward, senderDetails, recipientDetails, getCorrelationAuditData(jweRequest.getCorrelationId()), getCallAuditData(jweRequest.getApiCallId()), getParticipantCtxAuditData(jweRequest.getSenderCode(), jweRequest.getRecipientCode(), jweRequest.getCorrelationId()), path);
                 } else {
+                    if (!path.contains("on_")) {
```

Comment:
Give proper message that plain json body is valid for only on_action calls

Reply:
okay